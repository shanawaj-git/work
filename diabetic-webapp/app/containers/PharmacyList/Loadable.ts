/**
 * Asynchronously loads the component for PhamcyList screen
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
