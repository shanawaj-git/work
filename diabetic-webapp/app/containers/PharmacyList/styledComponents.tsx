import styled from 'styled-components';

export const Container = styled.div`
  padding: 20px;
  background-color: #fafafa;
  display: flex;
  flex-direction: column;
  width: 100%;
  .spinner-animation {
    align-items: center;
    justify-content: center;
    width: 35px;
    height: 35px;
  }
  .styled-spinner {
    display: flex;
    margin-top: 36px;
    justify-content: center;
  }
`;
export const Wrapper = styled.div`
  display: flex;
  padding-top: 50px;
  padding-bottom: 48px;
  justify-content: flex-start;
  align-items: fles-start;
  flex-direction: column;
  width: 100%;
`;
