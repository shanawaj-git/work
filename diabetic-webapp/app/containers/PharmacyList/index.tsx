import React, { useEffect, useState } from 'react';
import { Container, Wrapper } from './styledComponents';
import { Spinner } from '@albathanext/design-system';
import PharmacySelectionTabs, {
  PharmacySelectionTab,
} from './../../components/organism/PharmacySelectionTabs';

function PharmacyList() {
  // TODO: To be changed with API Integration
  const [loader, setLoader] = useState(true);
  useEffect(() => {
    const timer = setTimeout(() => {
      setLoader(false);
    }, 3000);
    return () => clearTimeout(timer);
  });
  return (
    <Container>
      <Wrapper>
        <PharmacySelectionTabs activeTab={PharmacySelectionTab.Delivery} />
      </Wrapper>
      <div className="styled-spinner">
        {loader ? <Spinner /> : <h1>Comming soon</h1>}
      </div>
    </Container>
  );
}

export default PharmacyList;
