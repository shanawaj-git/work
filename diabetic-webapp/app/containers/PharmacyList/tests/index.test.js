import React from 'react';
import { render } from 'react-testing-library';
import { BrowserRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import PharmacyList from '../index';

const MockPharmacyList = () => (
  <BrowserRouter>
    <IntlProvider locale="en">
      <PharmacyList />
    </IntlProvider>
  </BrowserRouter>
);
describe('<PharmacyList />', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<MockPharmacyList />);
    expect(firstChild).toMatchSnapshot();
  });
  // TODO: change test implementation after integrating with API
  it('should render a list of pharmacies', () => {
    render(<MockPharmacyList />);
    jest.runAllTimers();
    const pharmacies = document.querySelector('.styled-spinner');
    expect(pharmacies.childNodes.length).toBeGreaterThanOrEqual(0);
  });
});
