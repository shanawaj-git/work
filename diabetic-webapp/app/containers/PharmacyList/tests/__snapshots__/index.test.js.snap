// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PharmacyList /> should render and match the snapshot 1`] = `
<div
  class="sc-bdVaJa fEgtcV"
>
  <div
    class="sc-bwzfXH bjWyWH"
  >
    <div
      class="sc-htpNat jTUQWS"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 sc-bxivhb eGazJq css-1jfg05l-MuiTypography-root"
      >
        Select Pharmacy
      </h2>
      <div
        class="root MuiTabs-root styledComponents__StyledTabs-sc-16g4axo-1 gllqKx tabs css-1ujnqem-MuiTabs-root"
        id="pharmacy-selection-tabs"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="flexContainer MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected styledComponents__StyledTab-sc-16g4axo-0 iqlSvE css-1ujykiq-MuiButtonBase-root-MuiTab-root"
              data-testid="delivery"
              id="pharmacy-selection-tabs-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              Delivery
              <span
                class="indicator MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth styledComponents__StyledTab-sc-16g4axo-0 gcizIM css-1ujykiq-MuiButtonBase-root-MuiTab-root"
              data-testid="clickncollect"
              id="pharmacy-selection-tabs-tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Click & Collect
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="styled-spinner"
  >
    <img
      alt=""
      class="spinner-animation"
      src="IMAGE_MOCK"
    />
  </div>
</div>
`;
